### PR TITLE
Rename fedrpc to xrpc; implement xrpc and xrpc-server packages

### PR DESCRIPTION
**Pull request by pfrazee**
- Renames fedrpc to xrpc
- Implements xrpc (client) and xrpc-server (server) packages
- Still TODO
  - Implement server routes
  - Implement client (and server?) code generation

### CODE REVIEW

File: packages/xrpc-server/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    xrpc: 1,
+    id: 'io.example.ping',
+    type: 'query',
+    parameters: { message: { type: 'string' } },
+    output: {
+      encoding: 'text/plain',
+    },
+  }
+])
+server.method('io.example.ping', (params: xrpcServer.Params) => {
```
Comment (dholms) [inline packages/xrpc-server/README.md:19]: oh this is cool

File: packages/xrpc-server/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@adxp/xrpc-server",
```
Comment (dholms) [inline packages/xrpc-server/package.json:2]: make sure to add both of these pkgs to the root `tsconfig.json`

File: packages/xrpc/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    if (!schema) {
+      throw new Error(`Method schema not found: ${methodNsid}`)
+    }
+    const httpMethod = getMethodSchemaHTTPMethod(schema)
+    const httpUri = constructMethodCallUri(schema, this.uri, params)
+    const httpHeaders = constructMethodCallHeaders(schema, opts)
+
+    let res
+    try {
+      res = await axios({
```
Comment (dholms) [inline packages/xrpc/src/client.ts:87]: im not opposed, because I ❤️ axios. but is it actually buying us anything here?

**Review by dholms — APPROVED** [review]
this is looking really good. love how it's shaping up 🙏

File: packages/xrpc-server/package.json (hunk 2 above)
Comment (pfrazee) [reply packages/xrpc-server/package.json:2]: Done

File: packages/xrpc/src/client.ts (hunk 3 above)
Comment (pfrazee) [reply packages/xrpc/src/client.ts:87]: Convenient platform universality?

File: packages/xrpc/src/util.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  } else if (status >= 400 && status < 500) {
+    resCode = ResponseType.InvalidRequest
+  } else {
+    resCode = ResponseType.InternalServerError
+  }
+  return resCode
+}
+
+export function httpResponseBodyParse(
+  mimeType: string,
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply packages/xrpc/src/util.ts]: 👍

**Pull request merged**